test/NuGet.Core.Tests/NuGet.Protocol.Tests/Resources/RepositorySignatureResourceTests.cs

```diff
@@ -88,6 +89,71 @@ public void RepositorySignatureResource_NoCertInfo()
             Assert.Throws<FatalProtocolException>(() => new RepositorySignatureResource(jobject, repo));
         }
 
+        [Fact]
+        public async Task RepositorySignatureResource_RepositorySignatureInfo()
+        {
+            // Arrange
+            var source = $"http://unit.test/{Guid.NewGuid()}/v3-with-flat-container/index.json";
+            var responses = new Dictionary<string, string>();
+            responses.Add(source, JsonData.repoSignIndexJsonData);
+            responses.Add("https://api.nuget.org/v3-index/repository-signatures/index.json", JsonData.repoSignData);
+
+            var repo = StaticHttpHandler.CreateSource(source, Repository.Provider.GetCoreV3(), responses);
+
+            // Act
+            var repositorySignatureResource = await repo.GetResourceAsync<RepositorySignatureResource>();
+            repositorySignatureResource.UpdateRepositorySignatureInfo();
+
+            // Assert
+
+            var repositorySignatureInfo = RepositorySignatureInfoProvider.Instance.GetRepositorySignatureInfo(source);
```

Comment:
Why not use `repo.GetResourceAsync<RepositorySignatureInfo>(source)` instead?  Much cleaner than a static singleton.